### ISSUE TITLE
Return more detailed error codes from server that can be reacted on by the game clients

### ISSUE THREAD

**Issue by hdjay0129** [heroiclabs/nakama#277]
## Description
Currently nakama2 server doesn't return detailed error codes like nakama1 to the clients. 
Returning specific error codes would be useful for doing better client side error handling.

**Comment by novabyte** [comment]
@hdjay0129 We return both GRPC and HTTP status codes at the moment. What other codes would be useful?

**Comment by hdjay0129** [comment]
@novabyte Yes, I see that API's now return HTTP, GRPC status codes with ApiResponseException and WebSocket status codes with WebSocketsException.
But some cases require, special codes. Earlier Nakama1 had internal error codes mapping which were returned with the API calls. 
For instance now, I can't decide why MatchJoin call failed was it due to invalid match id or the match i'm trying to join no longer exists on server.
Also, With the MatchDataSend, when send match data to no longer existing matches or with invalid match id, don't get any error at all on client side saying your match data send call failed or for what reason it failed.

**Comment by mofirouz** [comment]
@hdjay0129 I see what you mean. This rises another follow up question I think. 

What would you do differently in case of an Invalid Match ID error vs Match that doesn't exist?

**Comment by hdjay0129** [comment]
@mofirouz , I wouldn't do anything differently for such a case. I would just stop connecting to match if it no longer exists and show error popup according to that match no longer exists or match id is incorrect.

Our case, specially is that, We store player's last joined match_id in his metadata so that the user can  re-join the match in case sudden disconnect or app close. So, While trying to re-join the match can no longer exists and the player will get an WebSocketsException error message saying "Match Not Found" for which we now check for exact string matching with error message.

I think having Nakama's internal error codes(as it was with Nakama1), will make such cases more clearer for developers to handle at client side.
